src/pages/TaskDetailsPage.js

```diff
@@ -1,43 +1,104 @@
-import React, { useState } from 'react';
-import { makeStyles } from '@material-ui/styles';
-import { useSelector } from 'react-redux';
+import React, { useState, useEffect } from 'react';
+import { connect } from 'react-redux';
+import { injectIntl } from 'react-intl';
+import { bindActionCreators } from 'redux';
+import { withTheme, withStyles } from '@material-ui/core/styles';
 import {
-  Form, Helmet, useTranslations, useModulesManager, useHistory,
+  Form, Helmet,
+  withHistory,
+  withModulesManager,
+  formatMessage,
 } from '@openimis/fe-core';
 import TaskHeadPanel from '../components/TaskHeadPanel';
 import TaskPreviewPanel from '../components/TaskPreviewPanel';
+import { fetchTask, updateTask } from '../actions';
 
-const useStyles = makeStyles((theme) => ({
+const styles = (theme) => ({
   page: theme.page,
-}));
-
-function TaskDetailsPage() {
-  const rights = useSelector((store) => store.core.user.i_user.rights ?? []);
-  const classes = useStyles();
-  const modulesManager = useModulesManager();
-  const history = useHistory();
-  const { formatMessage } = useTranslations('tasksManagement', modulesManager);
+});
+
+function TaskDetailsPage({
+  intl,
+  rights,
+  taskUuid,
+  task,
+  history,
+  modulesManager,
+  classes,
+  fetchTask,
+  updateTask,
+}) {
   const [editedTask, setEditedTask] = useState({});
   const back = () => history.goBack();
 
+  useEffect(() => {
+    if (taskUuid) {
+      fetchTask(modulesManager, [`id: "${taskUuid}"`]);
+    }
+  }, [taskUuid]);
+
+  useEffect(() => {
+    if (task) {
+      setEditedTask(task);
+    }
+  }, [task]);
+
+  const isMandatoryFieldsEmpty = () => !editedTask?.taskGroup;
+
+  const canSave = () => !isMandatoryFieldsEmpty();
```

Comment:
Here it could be good to make the 'save' icon blocked of nothing is changed. It should be active once something is changed on form level.

Reply:
done